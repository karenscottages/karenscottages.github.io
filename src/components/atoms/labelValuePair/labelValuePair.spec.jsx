import React from 'react';
import renderer from 'react-test-renderer';
import { LabelValuePair, LabelCheckBoxPair } from './';

describe("Atoms/LabelValuePair", () => {
    describe("LabelValuePair", () => {
        it('Renders with label and text', () => {
            const tree = renderer.
                create(<LabelValuePair label="Test label">Example Value</LabelValuePair>).
                toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe("LabelValuePair", () => {
        it('Renders with label, text and class', () => {
            const tree = renderer.
                create(<LabelValuePair label="Test label" className="is-2">Example Value</LabelValuePair>).
                toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe("LabelValuePair", () => {
        it('Renders with label, text and 2 class values', () => {
            const tree = renderer.
                create(<LabelValuePair label="Test label" className="is-2 has-text-weight-bold">Example Value</LabelValuePair>).
                toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe("LabelCheckBoxPair", () => {
        it('Renders with checkbox and label', () => {
            const tree = renderer.
                create(<LabelCheckBoxPair label="Test label"/>).
                toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe("LabelCheckBoxPair", () => {
        it('Renders with checkbox, label and onClick', () => {
            const tree = renderer.
                create(<LabelCheckBoxPair label="Test label" onClick={() => {}}/>).
                toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe("LabelCheckBoxPair", () => {
        it('Renders with checkbox, label and className', () => {
            const tree = renderer.
                create(<LabelCheckBoxPair label="Test label" className="ml-4"/>).
                toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});