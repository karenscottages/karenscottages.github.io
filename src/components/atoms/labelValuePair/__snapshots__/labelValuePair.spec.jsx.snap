// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/LabelValuePair LabelCheckBoxPair Renders with checkbox and label 1`] = `
<label
  className="checkbox"
>
  <input
    type="checkbox"
  />
  <span
    className="ml-2"
  >
    Test label
  </span>
</label>
`;

exports[`Atoms/LabelValuePair LabelCheckBoxPair Renders with checkbox, label and className 1`] = `
<label
  className="checkbox"
>
  <input
    type="checkbox"
  />
  <span
    className="ml-4"
  >
    Test label
  </span>
</label>
`;

exports[`Atoms/LabelValuePair LabelCheckBoxPair Renders with checkbox, label and onClick 1`] = `
<label
  className="checkbox"
  onClick={[Function]}
>
  <input
    type="checkbox"
  />
  <span
    className="ml-2"
  >
    Test label
  </span>
</label>
`;

exports[`Atoms/LabelValuePair LabelValuePair Renders with label and text 1`] = `
<div
  className="columns is-multiline"
>
  <div
    className="column "
  >
    <label
      className="label"
    >
      Test label
    </label>
  </div>
  <div
    className="column"
  >
    Example Value
  </div>
</div>
`;

exports[`Atoms/LabelValuePair LabelValuePair Renders with label, text and 2 class values 1`] = `
<div
  className="columns is-multiline"
>
  <div
    className="column is-2 has-text-weight-bold"
  >
    <label
      className="label"
    >
      Test label
    </label>
  </div>
  <div
    className="column"
  >
    Example Value
  </div>
</div>
`;

exports[`Atoms/LabelValuePair LabelValuePair Renders with label, text and class 1`] = `
<div
  className="columns is-multiline"
>
  <div
    className="column is-2"
  >
    <label
      className="label"
    >
      Test label
    </label>
  </div>
  <div
    className="column"
  >
    Example Value
  </div>
</div>
`;
